apache-rat-core/src/main/java/org/apache/rat/walker/Walker.java

```diff
@@ -33,38 +34,32 @@ public abstract class Walker implements IReportable {
     protected final File file;
     protected final String name;
 
-    protected final FilenameFilter filter;
+    protected final FilenameFilter filesToIgnore;
 
     protected static FilenameFilter regexFilter(final Pattern pattern) {
         return (dir, name) -> {
             final boolean result;
             if (pattern == null) {
-                result = true;
+                result = false;
             } else {
-                result = !pattern.matcher(name).matches();
+                result = pattern.matcher(name).matches();
```

Comment:
Thanks for getting rid of the negation. Has there been an error in here before or should line 42 be changed as well?

BEFORE
if(pattern==null) return true
else 
!pattern.matches()

NOW
if(pattern==null) return false
else 
pattern.matches()

or did I not get what is going on here?